renku/core/utils/templates.py

```diff
@@ -0,0 +1,871 @@
+            )
+
+        template = Template(file.read_text(), keep_trailing_newline=True)
+        rendered_content = template.render(template_metadata)
+
+        render_path = render_base / relative_path
+        render_path.parent.mkdir(parents=True, exist_ok=True)
+
+        if action == "initialize":
+            _render_file_for_initialize(
+                render_path=render_path,
+                path_in_project=path_in_project,
+                relative_path=relative_path,
+                rendered_content=rendered_content,
+            )
+        elif action == "set":
+            _render_file_for_set(
+                render_path=render_path,
+                path_in_project=path_in_project,
+                relative_path=relative_path,
+                rendered_content=rendered_content,
+                interactive=interactive,
+            )
+        elif action == "update":
+            _render_file_for_update(
+                render_path=render_path,
+                path_in_project=path_in_project,
+                relative_path=relative_path,
+                rendered_content=rendered_content,
+                checksums=checksums,
+                interactive=interactive,
+                immutable_template_files=immutable_template_files,
+            )
+        else:
+            raise errors.ParameterError(f"Invalid template render action: {action}")
+
+        checksum = hash_file(render_path) or hash_file(path_in_project)
+        if checksum is not None:
+            new_checksums[relative_path] = checksum
+
+    return render_base, new_checksums
+
+
+def _render_file_for_initialize(
+    render_path: Path,
+    path_in_project: Path,
+    relative_path: str,
+    rendered_content: str,
+):
+    def should_append(path: str):
+        return path.lower() in TEMPLATE_INIT_APPEND_FILES
+
+    def should_keep(path: str):
+        return path.lower() in TEMPLATE_KEEP_FILES
+
+    if not path_in_project.exists():
+        communication.echo(f"Initializing {relative_path} ...")
+        render_path.write_text(rendered_content)
+    elif should_append(relative_path):
+        communication.echo(f"Appending to {relative_path} ...")
+        render_path.write_text(path_in_project.read_text() + "\n" + rendered_content)
+    elif should_keep(relative_path):
+        communication.echo(f"Keeping file {relative_path} ...")
+    else:
+        communication.echo(f"Overwriting {relative_path} ...")
+        render_path.write_text(rendered_content)
+
+
+def _render_file_for_set(
+    render_path: Path,
+    path_in_project: Path,
+    relative_path: str,
+    rendered_content: str,
+    interactive: bool,
+):
+    """Decide what to do with files in a project when setting a template."""
+
+    def should_keep(path: str):
+        return path.lower() in TEMPLATE_KEEP_FILES
+
+    file_exists = path_in_project.exists()
+
+    if not file_exists:
```

Comment:
Above we have `if not path_in_project.exists():` and `file_exists` is not used other than here, we should make this consistent.